src/Starlight/Physics.hs

```diff
@@ -71,64 +73,71 @@ hit :: (Has (Reader (Seconds Float)) sig m, Has (Reader (System StateVectors)) s
 hit i c = do
   dt <- ask
   foldl' (go dt) c <$> view (beams_ @StateVectors) where
-  go (Seconds dt) char@Character{ actor = Actor{ position = c }, ship = Ship{ scale = r } } Beam{ angle = theta, position = o, firedBy = i' }
+  go (Seconds dt) char@Character{ actor = Actor{ position = c } } Beam{ angle = theta, position = o, firedBy = i' }
     | i /= i'
-    , intersects (c^._xy) (pure r) (o^._xy) (cartesian2 (pure <$> theta) 1)
+    , intersects (c^._xy) (char^.actor_.magnitude_ * 0.5) (o^._xy) (cartesian2 (pure <$> theta) 1)
     = char & ship_.armour_.min_.coerced -~ (damage * dt)
     | otherwise
     = char
   damage = 100 :: Float
 
 
 runActions
-  :: ( Effect sig
-     , Has (Reader (Seconds Float)) sig m
+  :: ( Has (Reader (Seconds Float)) sig m
      , Has (Reader (System StateVectors)) sig m
      , Has (State (System Body)) sig m
+     , HasCallStack
      )
   => CharacterIdentifier
   -> Character
   -> m Character
 runActions i c = do
   dt <- ask @(Seconds Float)
   system <- ask @(System StateVectors)
-  execState c (traverse_ (go dt system) (actions c)) where
-  go dt system = \case
-    Thrust -> do
-      rotation <- use (actor_.rotation_)
-      actor_ %= applyForce ((thrust ^*) <$> rotate rotation (unit _x)) dt
+  foldM (go dt system) c (actions c) where
```

Comment:
Back to writing this as a fold. Shoving the character in a `State` cell just for the sake of the one loop was pretty pointless.